adr/0008-phasing-out-config-classes.adoc

```diff
@@ -0,0 +1,110 @@
+On top of that, many of the new features were more straightforward to implement in this new model.
+
+A massive effort has been made to migrate from the legacy approach to the recommended approach, and all the Quarkus Core extensions have been switched to the `@ConfigMapping` approach.
+There are a few legacy classes left but they are there for compatibility purposes, for some config properties massively consumed outside of Quarkus that we didn't want to break to smooth the transition to the new model.
+
+At this stage, it is time to start planning phasing out the legacy approach.
+By phasing out, we mean that the approach with `@ConfigRoot` annotated **classes** will no longer be functional, including for external extensions.
+
+With the Quarkus core extension having switched to the new model, we have many examples we can point people to, and we are also very confident in the robustness of the `@ConfigMapping` support from SmallRye Config.
+
+@radcortez has also improved the performance of the new approach and allowed it to scale better.
+
+The main goal of this ADR is to propose an approach for the transition that reduces friction.
+
+== Why it is essential
+
+- We reduce the code base complexity and avoid maintaining the legacy code forever. It is complex, and the team's bandwidth is limited.
+- We avoid ambiguities and bad practices.
+- We avoid bugs that have never been fixed in the legacy approach due to limited bandwidth or high complexity.
+- We make sure we build the config on a unified approach making the config code more portable (from applications to extensions or even outside of Quarkus).
+- We reduce our startup footprint (with marginal improvements in terms of RSS and startup time).
+
+== Potential difficulties
+
+=== Transitioning
+
+Transitioning to config interfaces is a well-known and straightforward process.  For example, https://github.com/quarkusio/quarkus/pull/46070/files[this PR] illustrates such a transition in a non-simple extension (the MongoDB client extension).
+It can be a bit tedious, though, for extensions with massive configurations or projects with a lot of extensions (Camel Quarkus comes to mind).
+
+The main difficulty we identified is that the configuration is mutated in some extensions. It was possible with the legacy approach but never recommended. The new approach enforces the behavior, as the configuration is now immutable.
+It should concern very few extensions. Also, we need to stop this harmful practice anyway.
+
+=== Ecosystem
+
+The Quarkus extension ecosystem is broad and diverse: some extensions are very well maintained and following the progress of Quarkus, some are lagging and adjusted when time permits.
+
+If we phase out the legacy approach, we need the extension ecosystem ready.
+Indeed, each extension must be updated and released, as the legacy mechanism will be sunsetted.
+
+For Quarkiverse extensions, the Quarkus community can be proactive as we can push changes there if needed.
```

Comment:
didn't we submit prs for those we could find?
How about camel and others in platform ? 

if we have them submitted lets link to them.

Reply:
I can adjust the wording as I adjusted most if not all of them after I wrote this.

But I see no point in pointing to all the PRs in the ADR.